url9a01f43c15_poragchowdhury_securitygame_tgz-pJ8-securitygame_GameMasterJ8/src/securitygame/Network.java

```diff
@@ -0,0 +1,533 @@
+	public void printHiddenNetwork()
+	{
+		ArrayList<Node> captured = getCapturedNodes();
+		//System.out.println("Captured: " + captured.size());
+		PrintWriter writer;
+		try {
+			writer = new PrintWriter(fullGraphName + "-hidden.graph", "UTF-8");
+			for (int i = 0; i < nodes.length; i++)
+			{
+				Node node = getNode(i);
+				if (node.isCaptured() == true){
+					int neighborSize = node.neighbor.size();
+
+					int neighborCounter = 0;
+					for(Node neighbor: node.neighbor)
+					{
+						if(neighbor.getNodeID()!=node.getNodeID())
+						{
+							if(neighborCounter==neighborSize-1)
+								writer.print(neighbor.getNodeID());
+							else
+								writer.print(neighbor.getNodeID()+",");
+						}
+						neighborCounter++;
+					}
+					writer.println();
+				}
+				else
+					writer.println("-1");
+			}
+			for (int i = 0; i < nodes.length; i++){
+				Node node = getNode(i);
+				if(node.isCaptured() == true)
+					writer.println(node.getPv()+","+node.getSv()+","+node.getHoneyPot());
+				else
+					writer.println("-1,-1,-1");
+			}
+			writer.close();
+		}
+		catch (FileNotFoundException e)
+		{
+			e.printStackTrace();
+		}
+		catch ( Exception e)
+		{
+			e.printStackTrace();
+		}
+	}
+
+	/**
+     * Print network in a file
+     */
+	public void printNetwork()
+	{
+		PrintWriter writer;
+		try {
+			writer = new PrintWriter(fullGraphName + ".graph", "UTF-8");
+			for (int i = 0; i < nodes.length; i++)
+			{
+				Node node = getNode(i);
+				int neighborSize = node.neighbor.size();
+				int neighborCounter = 0;
+
+				if (node.neighbor.get(0) == null)
+				{
+					writer.print("-1");
+				}
+				else
+				{
+					for(Node neighbor: node.neighbor)
+					{
+						if(neighbor.getNodeID()!=node.getNodeID())
+						{
+							if(neighborCounter==neighborSize-1)
+								writer.print(neighbor.getNodeID());
+							else 
+								writer.print(neighbor.getNodeID()+",");
+						}
+						neighborCounter++;
+					}
+				}
+				writer.println();
+			}
+			for (int i = 0; i < nodes.length; i++)
+			{
+				Node node = getNode(i);
+				//writer.println(node.getPv()+","+node.getSv()+","+node.getHoneyPot());
+                writer.println(node.getPv()+","+node.getSv()+","+node.getHoneyPot());
+			}
+			writer.close();
+		}
+		catch (FileNotFoundException e)
+		{
+			e.printStackTrace();
+		}
+		catch ( Exception e)
+		{
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+     * Shuffles all the nodes in the network
+     */
+	public void shuffleNetwork()
+	{
+		ArrayList<Integer> assigned = new ArrayList<Integer>();
+		Random rand = new Random();
+		for(int i = 0; i< this.nodes.length; i++)
+		{
+			
+			while(true)
+			{
+				int id = rand.nextInt(nodes.length);
+				if((assigned.size()==0) || (!assigned.contains(id)))
+				{
+					this.nodes[i].setNodeID(id);
+					assigned.add(id);
+					break;
+				}
+				
+			}
+		}
+	}
+
+	/**
+     * Generates a random network based on the parameter class and prints it in a file
+     */
+	public void generateNetwork()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 51. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 318-503 into a separate method.